src/components/header/nav.tsx

```diff
@@ -0,0 +1,64 @@
+import { Box, Link } from "@mui/material";
+import { Auth } from "../../atoms/auth";
+
+interface HomeNavProps { auth: Auth | undefined}
+
+const HomeNav = (props: HomeNavProps) => {
+    const { auth } = props;
+
+    interface NavButtonProps {
+        text: string, 
+        selected: boolean, 
+        sx_props: object
+    } 
+    const NavButton = (props: NavButtonProps) => {
```

Comment:
If you destrucutred the props in the function params you could then set the function as an implicit return